### PR TITLE
feat(core): add three-way comparison operator for FixString class

### PR DESCRIPTION
**Pull request by dkrutskikh**
### **PR Type**
Enhancement

___

### **Description**
- Add three-way comparison operator (`<=>`) for FixString class

- Support comparisons with FixString, StringLike, and C strings

- Include comprehensive test coverage for spaceship operator

- Update documentation to reference new comparison operators

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["FixString Class"] --> B["operator<=> declarations"]
  B --> C["FixString vs FixString"]
  B --> D["FixString vs StringLike"]
  B --> E["FixString vs C string"]
  C --> F["Implementation with std::strong_ordering"]
  D --> F
  E --> F
  F --> G["Comprehensive test suite"]
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>fix_string.hpp</strong><dd><code>Add spaceship operator declarations</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

include/core/fix_string.hpp

<ul><li>Add five new <code>operator<=></code> function declarations<br> <li> Support comparisons between FixString objects of different sizes<br> <li> Support comparisons with StringLike objects and C strings<br> <li> Update existing operator== documentation with cross-references</ul>

</details>

  </td>
  <td><a href="https://github.com/ToymanInteractive/toygine2/pull/101/files#diff-2ef8f9f195db23c05d195ee56d0aaa0c714eeaf911132af27804a48fa895ce44">+142/-1</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>fix_string.inl</strong><dd><code>Implement spaceship operator logic</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

include/core/fix_string.inl

<ul><li>Implement five <code>operator<=></code> template functions<br> <li> Use <code>std::memcmp</code> for runtime and <code>cstrcmp</code> for compile-time comparisons<br> <li> Handle empty string cases and size differences properly<br> <li> Include null pointer assertions for C string comparisons</ul>

</details>

  </td>
  <td><a href="https://github.com/ToymanInteractive/toygine2/pull/101/files#diff-a47dfdd2a0becc17f40499abd90d00211a258ffaea3ce144fd5fc11d5ad08afd">+139/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>core.hpp.in</strong><dd><code>Refine StringLike concept definition</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/core/core.hpp.in

<ul><li>Update <code>StringLike</code> concept to use <code>std::same_as</code> instead of <br><code>std::convertible_to</code><br> <li> Rename template parameter from <code>s</code> to <code>str</code> for consistency</ul>

</details>

  </td>
  <td><a href="https://github.com/ToymanInteractive/toygine2/pull/101/files#diff-ebeff679f3b46ed6304ddd3b447e05c60fcfa5f7c5e989efc2b3598afb8ce16e">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>fix_string_test.cpp</strong><dd><code>Add spaceship operator tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

test/core/fix_string_test.cpp

<ul><li>Add comprehensive test case for <code>operator<=></code><br> <li> Test all comparison combinations (FixString, StringLike, C strings)<br> <li> Include edge cases: empty strings, Unicode, special characters<br> <li> Verify both runtime and compile-time behavior</ul>

</details>

  </td>
  <td><a href="https://github.com/ToymanInteractive/toygine2/pull/101/files#diff-cc1ff39c1921067e1cd9ffcf0bc8c8f3d7349f0178f1eb6ddeac099bc0a2d2ae">+345/-0</a>&nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>Doxyfile.in</strong><dd><code>Include inline files in documentation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

docs/Doxyfile.in

<ul><li>Remove <code>*.inl</code> from <code>EXCLUDE_PATTERNS</code><br> <li> Allow documentation generation for inline implementation files</ul>

</details>

  </td>
  <td><a href="https://github.com/ToymanInteractive/toygine2/pull/101/files#diff-01ce6f949e9bb8f3a3edb6a4a55694867cf1b028cac8eb9ab6f15530a809d58c">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

File: include/core/fix_string.hpp [hunk 1]
```diff
@@ -2082,10 +2089,144 @@ template <std::size_t allocatedSize>
...
+
+  \param lhs The C string to compare.
+  \param rhs The FixString object to compare.
+
+  \return std::strong_ordering::less if \a lhs is lexicographically less than \a rhs, std::strong_ordering::equal if
+          they are equal, or std::strong_ordering::greater if \a lhs is lexicographically greater than \a rhs.
+
+  \pre The \a rhs pointer must not be null.
+
+  \note The comparison is case-sensitive.
```
[the remark this answers is not in the document]
Comment (dkrutskikh) [reply include/core/fix_string.hpp:2219]: fixed
